kedro_vertexai/config.py

```diff
@@ -127,18 +128,31 @@ class RunConfig(BaseModel):
     resources: Optional[Dict[str, ResourcesConfig]] = dict(
         __default__=ResourcesConfig(cpu="500m", memory="1024Mi")
     )
+    node_selectors: Optional[Dict[str, Dict[str, str]]] = {}
     dynamic_config_providers: Optional[List[DynamicConfigProviderConfig]] = []
     mlflow: Optional[MLFlowVertexAIConfig] = None
 
-    def resources_for(self, node):
-        if node in self.resources.keys():
-            result = self.resources["__default__"].dict()
-            result.update(
-                {k: v for k, v in self.resources[node].dict().items() if v is not None}
-            )
-            return result
-        else:
-            return self.resources["__default__"].dict()
+    def resources_for(self, node: str, tags: set = set()):
+        result = self.resources["__default__"].dict()
+        return self._create_config(node, tags, self.resources, result)
+
+    def node_selectors_for(self, node: str, tags: set = set()):
+        result = {}
+        return self._create_config(node, tags, self.node_selectors, result)
+
+    @staticmethod
+    def _create_config(node: str, tags: set, params: dict, result: dict):
```

Comment:
1. Make `result: dict` optional (set it to `None` and use `results = results or {}` in the function body) and also it would be better to rename it to `default_config`
2. I would rename this function to `_config_for` to have the same name convention